File: Movies2B/Backend/backendMovies/api/models/Movie.js
/**
 * Movie.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {

    name:{
      type: 'string',
      minLength: 3,
      maxLength: 60,
    },
    year: {
      type: 'string',
      minLength: 4,
    },
    price: {
      type: 'string',
      minLength: 3,
    },
    gender: {
      type: 'string',
      minLength: 3
    },
    sinopsis: {
      type: 'string',
      minLength: 10
    },
    duration: {
      type: 'string',
      minLength: 3
    },
    image: {
      type: 'string'
    },
    wasPurchased: {
      type: 'boolean'
    },
    bePurchased:{
      collection: 'purchase',
      via: 'fkMovie'
    }
  },

};

File: Movies2B/Movies/app/src/main/java/com/example/movies/Movie.kt
package com.example.movies

import android.util.Log

class Movie (
    var id: Int,
    var name: String,
    var year: String,
    var price: String,
    var gender: String,
    var sinopsis: String,
    var duration: String,
    var image: String
){

    companion object {
        var movies = arrayListOf<Movie>()
        private var adapter : MovieHttpAdapter = MovieHttpAdapter()

        fun refreshList(){
            //if (adapter.getAll() != null){
                movies = adapter.getAll()!!
            Log.i("load:", "movies loaded")
            //}

        }

    }

}